.github/workflows/sts-mock-push-to-main.yml

```diff
@@ -0,0 +1,250 @@
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: sts-mock
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@9a9194f87191a7e9055e3e9b95b8cfb13023bb08 #main
+        with:
+          submodules: true
+          fetch-depth: 0
+
+      - name: Setup nodeJS v20
+        uses: actions/setup-node@26961cf329f22f6837d5f54c3efd76b480300ace #main
+        with:
+          node-version: 20
+          cache: npm
+          cache-dependency-path: sts-mock/package-lock.json
+
+      - name: Install dependencies
+        run: npm install
+      
+      - name: Run Tests
+        run: npm run test:unit
+
+      - name: Run infra Tests
+        run: npm run test:infra
+
+      # Generate test coverage report for Sonar main branch analysis
+      - name: SonarCloud Scan
+        uses: sonarsource/sonarcloud-github-action@f5003fc9688ade81ce47b57a3fa97a8d3f12de4c #master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+  # build-and-push-test-image-to-dev:
+  #   name: Build and push test image to Dev
+  #   needs: sts-mock-tests-and-sonar-scan
+  #   runs-on: ubuntu-latest
+  #   env:
+  #     STS_MOCK_DEV_TEST_IMAGE_REPOSITORY_URI: ${{ secrets.STS_MOCK_DEV_TEST_IMAGE_REPOSITORY_URI }}      
+  #     DEV_CONTAINER_SIGN_KMS_KEY: ${{ secrets.DEV_CONTAINER_SIGN_KMS_KEY }}
+  #     IMAGE_TAG: latest
+  #   defaults:
+  #     run:
+  #       shell: bash
+  #       working-directory: sts-mock
+  #   steps:
+  #   - name: Check out repository code
+  #     uses: actions/checkout@9a9194f87191a7e9055e3e9b95b8cfb13023bb08 #main
+  #     with:
+  #       submodules: true
+  #       fetch-depth: 0
+
+  #   - name: Build test image
+  #     run: |
+  #       docker build -t $STS_MOCK_DEV_TEST_IMAGE_REPOSITORY_URI:$IMAGE_TAG . 
+
+  #   - name: Configure AWS credentials for DEV
+  #     uses: aws-actions/configure-aws-credentials@55f725fcb586ca16a5ed5b0d75d464defbfa831b #main
+  #     with:
+  #       aws-region: eu-west-2
+  #       role-to-assume: ${{ secrets.STS_MOCK_DEV_GH_ACTIONS_ROLE_ARN }}
+
+  #   - name: Login to Amazon ECR DEV
+  #     uses: aws-actions/amazon-ecr-login@a81a5945e74802f35ca53aa274a9e00436e6210e #main
+
+  #   - name: Push image to DEV
+  #     run: |
+  #       docker push $STS_MOCK_DEV_TEST_IMAGE_REPOSITORY_URI:$IMAGE_TAG
+
+  #   - name: Install Cosign
+  #     uses: sigstore/cosign-installer@4959ce089c160fddf62f7b42464195ba1a56d382 #main
+  #     with:
+  #       cosign-release: 'v1.9.0'
+
+  #   - name: Code sign the Docker image
+  #     id: cosign-image             
+  #     run: |
+  #       cosign sign --key awskms:///$DEV_CONTAINER_SIGN_KMS_KEY $STS_MOCK_DEV_TEST_IMAGE_REPOSITORY_URI:$IMAGE_TAG
+
+  build-and-upload-sam-artifact-to-dev:
+    name: Validate & upload S3 artifact to dev
+    runs-on: ubuntu-latest
+    # needs:
+    #   - build-and-push-test-image-to-dev
```

Comment:
What's the rationale for commenting this out? I think the SAM upload should always be dependent on the test image being pushed successfully otherwise we could have a situation where we promote a build without the latest tests

Reply:
We currently have no test image to push so this will fail.